controller/sharding/sharding.go

```diff
@@ -459,22 +459,18 @@ func getDefaultShardMappingData(replicas int) []shardApplicationControllerMappin
 }
 
 func GetClusterSharding(kubeClient kubernetes.Interface, settingsMgr *settings.SettingsManager, shardingAlgorithm string, enableDynamicClusterDistribution bool) (ClusterShardingCache, error) {
-	var replicasCount int
-	if enableDynamicClusterDistribution {
-		applicationControllerName := env.StringFromEnv(common.EnvAppControllerName, common.DefaultApplicationControllerName)
-		appControllerDeployment, err := kubeClient.AppsV1().Deployments(settingsMgr.GetNamespace()).Get(context.Background(), applicationControllerName, metav1.GetOptions{})
-		// if app controller deployment is not found when dynamic cluster distribution is enabled error out
-		if err != nil {
-			return nil, fmt.Errorf("(dynamic cluster distribution) failed to get app controller deployment: %w", err)
-		}
+	applicationControllerName := env.StringFromEnv(common.EnvAppControllerName, common.DefaultApplicationControllerName)
+	appControllerDeployment, err := kubeClient.AppsV1().Deployments(settingsMgr.GetNamespace()).Get(context.Background(), applicationControllerName, metav1.GetOptions{})
```

Comment:
The controller can be a Deployment or a StatefulSet. I am not sure if `enableDynamicClusterDistribution` must be true when using a deployment. If it is not possible, I don't think we should perform 2 k8s api calls for a warning log to help detect a misconfiguration, considering that 1 call will always fail.

Reply:
My issue was that I attempted sharding with enableDynamicClusterDistribution set to false, but the actual Deployment replica count was smaller than the sharding configuration. This caused a problem, and it was difficult to debug since no logs were recorded. Since the ArgoCD application controller typically runs as a Deployment, I believe it would be more accurate to retrieve the actual replica count using the Kubernetes API, even when enableDynamicClusterDistribution is set to false. This approach could help prevent discrepancies between the sharding configuration and the actual replica count, making debugging easier. Would this be a suitable implementation? I would appreciate any feedback or suggestions on this matter.